cozy/typecheck.py

```diff
@@ -62,6 +62,15 @@ def retypecheck(exp, env : {str : syntax.Type} = None, fenv = None):
 def is_numeric(t):
     return t in (INT, LONG, FLOAT)
 
+# TODO: complete it
+def is_constant(e):
+    if isinstance(e, syntax.ENum):
+        return True
+    elif isinstance(e, syntax.EUnaryOp):
+        return is_constant(e.e)
```

Comment:
What is the intended behavior of this function? I think you might want to restrict this case to unary minus on numbers.